src/frontends/paddle/tests/test_models/gen_scripts/generate_pool3d.py

```diff
@@ -0,0 +1,256 @@
+    data = np.arange(N * C * D * H * W).astype(data_type)
+    data_NCDHW = data.reshape(N, C, D, H, W)
+    data_NDHWC = data.reshape(N, D, H, W, C)
+
+    pooling_types = ["max", "avg"]
+
+    for i, pooling_type in enumerate(pooling_types):
+        # example 1:
+        # ceil_mode = False
+        paddle_attrs = {
+            # input=data_NCDHW, # shape: [2, 3, 4, 4, 4]
+            "pool_size": [3, 3, 3],
+            "pool_type": pooling_type,
+            "pool_stride": [3, 3, 3],
+            "pool_padding": [
+                1,
+                2,
+                1,
+            ],  # it is same as pool_padding = [1, 1, 2, 2, 1, 1]
+            "ceil_mode": False,
+            "exclusive": True,
+            "data_format": "NCDHW",
+        }
+        pool3d(pooling_type + "3d" + "Pool_test1", data_NCDHW, paddle_attrs)
+
+        # example 2:
+        # ceil_mode = True (different from example 1)
+        paddle_attrs = {
+            # input=data_NCDHW,
+            "pool_size": [3, 3, 3],
+            "pool_type": pooling_type,
+            "pool_stride": [3, 3, 3],
+            "pool_padding": [
+                [0, 0],
+                [0, 0],
+                [1, 1],
+                [2, 2],
+                [1, 1],
+            ],  # it is same as pool_padding = [1, 1, 2, 2, 1, 1]
+            "ceil_mode": True,
+            "exclusive": True,
+            "data_format": "NCDHW",
+        }
+        pool3d(pooling_type + "3d" + "Pool_test2", data_NCDHW, paddle_attrs)
+
+        # example 3:
+        # pool_padding = "SAME" (different from example 1)
+        paddle_attrs = {
+            # input=data_NCDHW,
+            "pool_size": [3, 3, 3],
+            "pool_type": pooling_type,
+            "pool_stride": [3, 3, 3],
+            "pool_padding": "SAME",
+            "ceil_mode": False,
+            "exclusive": True,
+            "data_format": "NCDHW",
+        }
+        pool3d(pooling_type + "3d" + "Pool_test3", data_NCDHW, paddle_attrs)
+
+        # example 4:
+        # pool_padding = "VALID" (different from example 1)
+        paddle_attrs = {
+            # input=data_NCDHW,
+            "pool_size": [3, 3, 3],
+            "pool_type": pooling_type,
+            "pool_stride": [3, 3, 3],
+            "pool_padding": "VALID",
+            "ceil_mode": False,
+            "exclusive": True,
+            "data_format": "NCDHW",
+        }
+        pool3d(pooling_type + "3d" + "Pool_test4", data_NCDHW, paddle_attrs)
+
+        # example 5:
+        # data_format = "NDHWC" (different from example 1)
+        paddle_attrs = {
+            # input=data_NDHWC, # shape: [2, 4, 4, 4, 3]
+            "pool_size": [3, 3, 3],
+            "pool_type": pooling_type,
+            "pool_stride": [3, 3, 3],
+            "pool_padding": [1, 2, 1],
+            "ceil_mode": False,
+            "exclusive": True,
+            "data_format": "NDHWC",
+        }
+        # NOT support data_format = "NDHWC" now
```

Comment:
why not support the NDHWC?

Reply:
The main reason is because maxpool does not directly provide `data_format` parameter.

It seems that two transpose ops need to be added to support

Reply:
Seems you have the idea to support it, can you implement it?

Reply:
Done, Now we support "NDHWC" input. And I add `paddle_avg3dPool_test10` to test the `padding` when `NDHWC` input

- https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/functional/max_pool3d_en.html